omreaderlib/src/Interface.scala

```diff
@@ -0,0 +1,61 @@
+package org.chipsalliance.t1.omreaderlib
+
+import scala.reflect.runtime.universe._
+import chisel3.panamalib.option._
+import chisel3.panamaom._
+import chisel3.panamaconverter.PanamaCIRCTConverter
+
+object OMReader {
+  def fromFile(mlirbcFile: os.Path, className: String): OMReader = {
+    new OMReader(os.read.bytes(mlirbcFile), className)
+  }
+
+  def fromBytes(mlirbc: Array[Byte], className: String): OMReader = {
+    new OMReader(mlirbc, className)
+  }
+}
+
+class OMReader private(mlirbc: Array[Byte], className: String) {
+  private val cvt = PanamaCIRCTConverter.newWithMlirBc(mlirbc)
+
+  private val pm = cvt.passManager()
+  assert(pm.populatePreprocessTransforms())
+  assert(pm.populateCHIRRTLToLowFIRRTL())
+  assert(pm.populateLowFIRRTLToHW())
+  assert(pm.populateLowHWToSV())
+  assert(pm.populateExportVerilog(_ => ()))
+  assert(pm.populateFinalizeIR())
+  assert(pm.run())
```

Comment:
I think we don't need to populate to Verilog here?
what we need to do is assume we have got the final mlirbc from firtool binary, evaluate and read from which.
depending on the populate function can be added in the future PR. e.g. remove firtool usage and switch to Scala.

Reply:
Populating these passes is essential. Currently T1 only produces mlirbc with `firtool --parse-only`.

Reply:
I have removed these since now we ran these passes via `firtool` before reading OM.